provider/azureProvider.js

```diff
@@ -227,64 +199,58 @@ class AzureProvider {
 
   getAdminKey () {
     const options = {
-      'host': functionAppName + constants.scmDomain,
-      'port': 443,
-      'path': constants.masterKeyApiApth,
-      'headers': {
-        'Authorization': constants.bearer + principalCredentials.tokenCache._entries[0].accessToken,
-        'Content-Type': constants.jsonContentType
+      url: `https://${functionAppName}${config.scmDomain}${config.masterKeyApiPath}`,
+      json: true,
+      headers: {
+        Authorization: config.bearer + principalCredentials.tokenCache._entries[0].accessToken
```

Comment:
As a future optimization, it might clean things up to use the modified `request` instance that `az-login` provides. That way we don't need to construct these authenticated `request` instances "manually".